File: iniciante/python/1116-dividindo-x-por-y.py
def entrada():
    numeros = list(map(int, input().split(' ')))
    n1 = numeros[0]
    n2 = numeros[1]
    return n1, n2


def dividindo(n):
    for i in range(n):
        dividendo, divisor = entrada()
        if divisor == 0:
            print('divisao impossivel')
        else:
            divisao = dividendo / divisor
            print(f'{divisao:.1f}')


numero_de_vezes = int(input())
dividindo(numero_de_vezes)

File: iniciante/c/1116-dividindo-x-por-y.c
#include <stdio.h>

int main(){
    int n_teste, i;
    float x, y, divisao;
    scanf("%d", &n_teste);
    for (i = 0; i < n_teste; i++){
        scanf("%f %f", &x, &y);
        if (y == 0.0){
            printf("divisao impossivel\n");
        }
        else{
            divisao = x / y;
            printf("%.1f\n", divisao);
        }
    }
    return 0;
}
